arcgis-ios-sdk-samples/Scenes/Get Elevation at a point/GetElevationPointViewController.swift

```diff
@@ -0,0 +1,122 @@
+    private func makeGraphics() {
+        graphicsOverlay.renderingMode = AGSGraphicsRenderingMode.dynamic
+        graphicsOverlay.sceneProperties?.surfacePlacement = AGSSurfacePlacement.relative
+        sceneView.graphicsOverlays.add(graphicsOverlay)
+    }
+    
+    private func makeScene() -> AGSScene {
+        let scene = AGSScene(basemapType: .imageryWithLabels)
+        
+        let surface = AGSSurface()
+        // Create an elevation source.
+        let elevationURL = URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")
+        let elevationSource = AGSArcGISTiledElevationSource(url: elevationURL!)
+        
+        // Add the elevation source to the surface.
+        surface.elevationSources.append(elevationSource)
+        scene.baseSurface = surface
+        
+        return scene
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["GetElevationPointViewController", "ElevationViewController"]
+    }
+}
+
+// MARK: - AGSGeoViewTouchDelegate
+extension GetElevationPointViewController: AGSGeoViewTouchDelegate {
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        if let relativeSurfacePoint = sceneView?.screen(toBaseSurface: screenPoint) {
+//            graphicsOverlay.graphics.removeAllObjects()
+            dismiss(animated: true)
+            
+            // Create the symbol at the tapped point.
+            let marker = AGSSimpleMarkerSceneSymbol(style: .sphere, color: .red, height: 100, width: 100, depth: 200, anchorPosition: .center)
+            let point = AGSPoint(x: relativeSurfacePoint.x, y: relativeSurfacePoint.y, spatialReference: .wgs84())
+            self.graphicsOverlay.graphics.add(AGSGraphic(geometry: point, symbol: marker))
+            
+            // Get the surface elevation at the surface point.
+            self.sceneView.scene?.baseSurface!.elevation(for: relativeSurfacePoint) { (results: Double, error: Error?) in
+                if let error = error {
+                    self.presentAlert(error: error)
+                } else {
+                    self.showPopover(elevation: results, popoverPoint: screenPoint)
+                }
+            }
+        }
+    }
+
+    private func showPopover(elevation: Double, popoverPoint: CGPoint) {
+        guard let controller = storyboard?.instantiateViewController(withIdentifier: "ElevationViewController") as? ElevationViewController else {
+                return
+        }
+        // Setup the controller to display as a popover.
+        controller.modalPresentationStyle = .popover
+        controller.loadViewIfNeeded()
+        controller.elevationLabel?.text? = "Elevation at tapped point: " + (String(elevation.rounded()) + String("m"))
```

Comment:
Please use a measurement formatter to convert the elevation to a string.

Reply:
Despite defining the unit as `UnitLength.meters` while using this method, the outcome appears as "mi" when running the sample. Do you know why?

Reply:
Yes. The default behavior of `MeasurementFormatter` is to localize "according to the preferences of the formatter’s `locale`". So users in the US would see the measurement in miles (converted from meters, of course) and users in Britain, for example, would see the measurement in kilometers. That's one of the advantages of using a measurement formatter and I think it makes sense for this sample. However, you can get the measurement in meters for everybody, if you must, by specifying `MeasurementFormatter.UnitOptions.providedUnit` as one of the unit options:

```swift
measurementFormatter.unitOptions = .providedUnit
```